src/event_handler.rs

```diff
@@ -504,63 +511,148 @@ pub async fn handle_event_logs(
-        let http_rpc_client = app_state.http_rpc_client.lock().unwrap().clone().unwrap();
-        let Ok(txn_receipt) = http_rpc_client
-            .get_transaction_receipt(pending_txn_hash)
-            .await
-        else {
-            resend_retries -= 1;
+    loop {
+        if pending_txns.lock().unwrap().is_empty() {
+            sleep(Duration::from_millis(200)).await;
             continue;
-        };
+        }
 
-        if txn_receipt.is_some() {
-            break;
+        let mut pending_txn_data = pending_txns.lock().unwrap().pop_front().unwrap();
+        let mut retry_txn = true;
+
+        while Instant::now() < pending_txn_data.retry_deadline && retry_txn == true {
+            sleep(Duration::from_secs(RESEND_TXN_INTERVAL)).await;
+
+            // Get the transaction receipt for the pending transaction to check if it's still pending or been dropped
+            let http_rpc_client = app_state.http_rpc_client.lock().unwrap().clone().unwrap();
+            let Ok(txn_receipt) = http_rpc_client
+                .get_transaction_receipt(pending_txn_data.txn_hash)
+                .await
+            else {
+                continue;
+            };
+
+            // Transaction is confirmed/mined and need not be resent
+            if txn_receipt.is_some() {
+                retry_txn = false;
+                break;
+            }
+
+            // Current txn is dropped/lost and need to be resent with higher gas price
+            pending_txn_data.gas_limit = pending_txn_data.gas_limit + GAS_LIMIT_BUFFER;
+            pending_txn_data.gas_price = U256::from(100 + RESEND_GAS_PRICE_INCREMENT_PERCENT)
+                * pending_txn_data.gas_price
+                / 100;
+
+            // Send the replacement txn if the current one is dropped or lost
+            while Instant::now() < pending_txn_data.retry_deadline {
+                let replacement_txn = pending_txn_data.txn_data.clone();
+                let replacement_txn = replacement_txn
+                    .nonce(pending_txn_data.nonce)
+                    .gas_price(pending_txn_data.gas_price)
+                    .gas(pending_txn_data.gas_limit);
+
+                let pending_txn = replacement_txn.send().await;
+                let Ok(pending_txn) = pending_txn else {
+                    let error_string = format!("{:?}", pending_txn.unwrap_err());
+
+                    // Retry after a delay if connection failed
+                    if error_string.contains("code: -32000")
+                        && (error_string.contains("connection") || error_string.contains("network"))
+                    {
+                        sleep(Duration::from_millis(200)).await;
+                        continue;
+                    }
+
+                    // Retry with a higher gas price if the replacement txn is underpriced
+                    if error_string.contains("code: -32000")
+                        && (error_string.contains("replacement transaction underpriced")
```

Comment:
really ? this error string would contain the "replacement" keyword?

Reply:
Yes I saw this is an error that pops up when the gas price for the replacement transaction is not sufficient enough. Though, will just check for "transaction underpriced" as it covers both error messages.